### PR TITLE
Add all widgets

### PR DESCRIPTION
**Pull request by aayush420**
#4 This will be a continuous PR. Will be adding widgets one by one. 
- [x] Accordion
- [x] Button
- [x] Card
- [x] Check
- [x] Icon
- [x] Label
- [x] Entry
- [x] Form
- [x] Hyperlink
- [x] List
- [x] Menu
- [x] Multi Line Entry
- [x] Password Entry
- [x] Progress Bar
- [x] Radio 
- [x] Select/Combo/Drop down
- [x] Separator
- [x] Slider
- [x] Table
- [x] Text Grid
- [x] Toolbar
- [x] Tree

...

### CODE REVIEW

**Comment by andydotxyz** [comment]
Great progress thanks.
For the list above, I don't think that FormItem is a widget - more a property of form.
And PopUp/ModalPopUp/PopUpMenu are overlays so may not be possible/sensible to add at this stage.

File: widget.go [hunk 1]
```diff
@@ -108,6 +138,332 @@ var widgets = map[string]widgetInfo{
...
+		},
+	},
+	"*widget.Select": {
+		name: "Select",
+		create: func() fyne.CanvasObject {
+			return widget.NewSelect([]string{"Option 1", "Option 2"}, func(value string) {})
+		},
+		edit: func(obj fyne.CanvasObject) []*widget.FormItem {
+			s := obj.(*widget.Select)
+			initialOption := widget.NewSelect(s.Options, func(opt string) { s.SetSelected(opt) })
```
Comment (andydotxyz) [inline widget.go]: This needs to support "(Select one)" which is the default - it shows as expected to start with but then is no longer available if you select an option.

**Review by andydotxyz — CHANGES_REQUESTED** [review]
This is so cool.
Just a minor change requested.

I also noticed that the new MultiLineEntry property fields don't seem to be multiline - but I cannot figure out why (try editing the text of a MultiLineEntry) - have you noticed this as well?

**Comment by aayush420** [comment]
> This is so cool.
> Just a minor change requested.
> 
> I also noticed that the new MultiLineEntry property fields don't seem to be multiline - but I cannot figure out why (try editing the text of a MultiLineEntry) - have you noticed this as well?

Yeah I did, and it's a bit weird. Also, the name on the properties palette is shown as `Entry` instead of `MultiLineEntry`. This might be happening because of the fact that I'm accessing the Entry object from the CanvasObject `obj` in here - 
image

**Comment by aayush420** [comment]
> Yeah I did, and it's a bit weird. Also, the name on the properties palette is shown as `Entry` instead of `MultiLineEntry`. This might be happening because of the fact that I'm accessing the Entry object from the CanvasObject `obj` in here -
> image

Just found out that doing `mle := obj.(*widget.Entry)` has caused the properties to come from the original entry's properties. Need to figure out a way to reference `MultiLineEntry`, but it isn't defined by widget.

**Comment by aayush420** [comment]
> 
> Just found out that doing `mle := obj.(*widget.Entry)` has caused the properties to come from the original entry's properties. Need to figure out a way to reference `MultiLineEntry`, but it isn't defined by widget.

The same problem is there with passwordEntry's properties palette too (as this was also called as `*widget.Entry`).

**Comment by aayush420** [comment]
> The same problem is there with passwordEntry's properties palette too (as this was also called as `*widget.Entry`).

Fixed this by patching the `choose` method to have special cases for `PasswordEntry` and `MultiLineEntry`

**Review by andydotxyz — APPROVED** [review]
Amazing thanks

**Pull request merged**